griptape/drivers/rerank/base_rerank_driver.py

```diff
@@ -0,0 +1,9 @@
+from abc import ABC, abstractmethod
+from attrs import define
+from griptape.artifacts import TextArtifact
+
+
+@define(kw_only=True)
+class BaseRerankDriver(ABC):
+    @abstractmethod
+    def run(self, query: str, artifacts: list[TextArtifact]) -> list[TextArtifact]: ...
```

Comment:
I struggled with this one for a bit but ultimately decided to go for plain lists. My reasoning is that `BaseRerankModule.run` returns `list[BaseArtifact]` since it should, in theory, be able to rerank any artifacts. As a downstream consequence, rerank drivers follow this pattern as well. I think there is no benefit in using a `ListArtifact` in the driver either because reranking is a simple operation of reordering of items and plain lists work find for that.